ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import {TagsInput} from '../../../shared/components/tags-input/tags-input';
+import {NamespaceFilter} from '../../../shared/components/namespace-filter';
+import {CheckboxFilter} from '../../../shared/components/checkbox-filter/checkbox-filter';
+import * as models from '../../../../models';
+
+require('./workflow-filters.scss');
+
+function getLabelsSuggestions(labels: Map<string, Set<string>>) {
+    const suggestions = new Array<string>();
+    Array.from(labels.entries()).forEach(([label, values]) => {
+        values.forEach(val => suggestions.push(`${label}=${val}`));
+    });
+    return suggestions;
+}
+
+interface WorkflowFilterProps {
+    workflows: models.Workflow[];
+    namespace: string;
+    phases: string[];
+    labels: string[];
+    onChange: (namespace: string, phases: string[], labels: string[]) => void;
+}
+
+interface WorkflowFilterState {
+    namespace: string;
+    phases: string[];
+    labels: string[];
+    error?: Error;
+}
+
+export class WorkflowFilters extends React.Component<WorkflowFilterProps, WorkflowFilterState> {
+    constructor(props: Readonly<WorkflowFilterProps>) {
+        super(props);
+        this.state = {
+            namespace: props.namespace,
+            phases: props.phases,
+            labels: props.labels
```

Comment:
Not too sure about this pattern... there should only be one source of truth about what the current `namespace`, `phases`, and `labels` are. Right now we are storing it in three different places: the URL, state in `workflows-list`, and this state. It should all be consolidated in one. Why store state here if you pass in an `onChange` that changes the state of the parent component... forcing a re-render here anyway?

Reply:
Realized the same thing, removed the state and used the props.